contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -1316,67 +1316,114 @@ pre_transform_target_entry(ResTarget *res, ParseState *pstate,
 			colname_start = pstate->p_sourcetext + res->name_location;
 		}
 
+		/*
+		 * Case 1 : Handle both singlebyte and multibyte aliases when delimited by 
+		 * square bracket(sqb) and double quoutes(dq).
+		 * For instance, queries like : SELECT 1 AS '您对“数据一览“中的车型，颜色，内饰，选装, ';
+		 * Case 2 : Preserve the case of aliases with ascii characters when there is no sqb and dq.
+		 * For instance, queries like: SELECT 1 AS ABCD;
+		 * Case 3 : Handle both singlebyte and multibyte aliases whose length is
+		 * more than or equals to 63 when not delimited by sqb and dq.
```

Comment:
Why don't we have to worry about singlebyte and multibyte aliases whose length is more than or equals to 63 when not delimited by sqb and dq?

Reply:
Aliases when delimited with sqb and dq are already handled in grammer rule. These type of aliases are truncated properly and also the case is preserved. In such case, we are just returning. So now we need to handle aliases which are not delimited with sqb and dq whose length is less than 63 and more than 63. In case of single byte aliases, we need to preserve the case while in the case of multibyte aliases if actual alias length is more than 63, we should retrieve the alias name after truncation.